vignettes/decorate-module-output.Rmd

```diff
@@ -0,0 +1,589 @@
+---
+title: "Customizing Module Output"
+author: "NEST CoreDev"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+vignette: >
+  %\VignetteIndexEntry{Customizing Module Output}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Introduction
+
+`teal` is a powerful `shiny`-based framework with built-in modules for interactive data analysis. 
+This document outlines the customization options available for modifying the output of `teal` modules. 
+You will learn how to use `teal_transform_module` to modify and enhance the objects created by `teal` modules, 
+enabling you to tailor the outputs to your specific requirements without rewriting the original module code.
+
+Adjusting input data or customizing module outputs in `teal` is accomplished using `transformators` created through 
+`teal_transform_module`. This document will focus on modifying outputs, and for clarity, we will refer to these 
+output-modifying objects as decoration objects or decorators.
+
+
+## Decorators
+
+### Server
+
+The simplest way to create a decorator is to use `teal_transform_module` without specifying a `ui`.
+This approach adds functionality solely to the server code of the `teal` app. 
+In the following example, we assume that the module contains an object (of class `ggplot2`) named `plot`. 
+We modify the title and x-axis label of plot:
+
+```{r}
+library(teal)
+static_decorator <- teal_transform_module(
+  server = function(id, data) {
+    moduleServer(id, function(input, output, session) {
+      reactive({
+        req(data())
+        within(data(), {
+          plot <- plot +
+            ggtitle("This is title") +
+            xlab("x axis")
+        })
+      })
+    })
+  }
+)
+```
+
+To simplify the repetitive elements of writing new decorators 
+(e.g., `function(id, data), moduleServer, reactive, within(data, ...)`), 
+you can use the `make_teal_transform_server(expr)` wrapper, which takes an `expression` as input:
```

Comment:
I would probably use a "convenience function" instead of a wrapper. Not a big deal though

Reply:
0f51ce57af086efc4e52b2bc2fd4e2b46f596399